pkg/controller/controller_instance.go

```diff
@@ -1696,6 +1844,7 @@ func (c *controller) processTerminalProvisionFailure(instance *v1beta1.ServiceIn
 	if failedCond == nil {
 		return fmt.Errorf("failedCond must not be nil")
 	}
+	c.removeInstanceFromRetryMap(instance)
```

Comment:
We should probably remove instance if `!shouldMitigateOrphan`. I guess we could just move this line inside `processProvisionFailure` under condition
https://github.com/kubernetes-incubator/service-catalog/blob/3837abf27fe5d9b090a7d530a81d6cebc634a79e/pkg/controller/controller_instance.go#L1752-L1755

Reply:
Which makes me wonder if we may just invoke `removeInstanceFromRetryMap` inside `clearServiceInstanceCurrentOperation` method and remove this invocation from all other places?

Reply:
>We should probably remove instance if !shouldMitigateOrphan. I guess we could just move this line inside processProvisionFailure under condition

There are cases where we do retries (and want backoff) without doing orphan mitigation in between (retrying a failed update, retrying a provisioning after a 400 or 403).  I'll do some additional review, but at this point I'm not convinced doing the `removeInstancefromRetryMap` within `clearServiceInstanceCurrentOperation` is the right place.

Reply:
`clearServiceInstanceCurrentOperation` is invoked a lot, I believe I saw it invoked as part of the orphan mitigation - from debug output, I am not comfortable doing the `removeInstanceFromRetryMap` in `clearServiceInstanceCurrentOperation` as I think it would prematurely clear out the instance.   @nilebox if you feel strongly about this let me know and I'll spend more time verifying.

Reply:
> I believe I saw it invoked as part of the orphan mitigation

@jboyd01 We only invoke `clearServiceInstanceAsyncOsbOperation` in case of orphan mitigation instead of `clearServiceInstanceCurrentOperation`. Orphan mitigation is considered part of the original Service Catalog operation.
Please check, I think it would make the code cleaner.

Reply:
@jboyd01 You're right, we do invoke `clearServiceInstanceCurrentOperation` from the `processDeprovisionSuccess` method even it was orphan mitigation. We also invoke it in the beginning of orphan mitigation from `recordStartOfServiceInstanceOperation` method. So it's safer to invoke `removeInstancefromRetryMap` explicitly like you do, agreed.